Warsaw_Tram/MapViewController.swift

```diff
@@ -47,46 +46,31 @@ class MapViewController: UIViewController {
         map.showsScale = true
     }
     
-    private func fetchTramsData() {
-        let stringURL : NSString = "https://api.um.warszawa.pl/api/action/wsstore_get?\(resourceId)&\(warsawTramsApiKey)"
-        let url : NSURL = NSURL(string: stringURL as String)!
-        let request = NSMutableURLRequest(URL: url)
-        request.HTTPMethod = "GET"
-        
-        let task = NSURLSession.sharedSession().dataTaskWithRequest(request) { data, response, error in
-            if error != nil {
-                self.goToMainDesktop(errorMessage: error!.localizedDescription)
-            } else if data != nil {
-                guard let jsonResult = (try! NSJSONSerialization.JSONObjectWithData(data!, options: NSJSONReadingOptions.MutableContainers)) as? NSDictionary,
-                    let result: NSArray = jsonResult["result"] as? NSArray else {
-                        return self.goToMainDesktop(errorMessage: "Brak wyników")
-                }
-                
-                for tram in result {
-                    if tram["Status"] as? String == "RUNNING" {
-                        if let line = tram["FirstLine"] as? String,
-                            let lat = tram["Lat"] as? CLLocationDegrees,
-                            let lon = tram["Lon"] as? CLLocationDegrees,
-                            let lowFloor = tram["LowFloor"] as? Bool {
-                                dispatch_async(dispatch_get_main_queue(), { 
-                                    self.addTramAnnotation(line, lat: lat, lon: lon, lowFloor: lowFloor)
-                                })
-                        }
-                    }
+    private func displayTramsOnMap() {
+        tramViewModel.getTramsData({ (trams) in
+            for tram in trams {
+                if tram.status == "RUNNING" {
```

Comment:
Instead of another level of nesting you can use `for tram in trams wehere tram.status == "RUNNING"`